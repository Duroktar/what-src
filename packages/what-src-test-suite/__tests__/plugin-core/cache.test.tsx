import * as T from '@what-src/plugin-core'

const service = T.getService({
  basedir: '/home/user/app/packages/',
  cache: {},
  options: {},
})

it('caches source locations', () => {
  const testArgs = [
    [{ col: 1, line: 2 }, 'filename1.ts'],
    [{ col: 4, line: 7 }, 'filename2.ts'],
    [{ col: 8, line: 1 }, 'filename3.ts'],
  ] as const

  const cacheKeys: string[] = []

  testArgs.forEach(test => {
    cacheKeys.push(service.cache(test[0], test[1]))
  })

  expect<T.SourceCache>(service.getSourceCache()).toMatchSnapshot()
})
