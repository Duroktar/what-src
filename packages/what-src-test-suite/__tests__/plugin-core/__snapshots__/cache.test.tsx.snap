// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`caches source locations 1`] = `
Object {
  "1": "{\\"filename\\":\\"filename1.ts\\",\\"line\\":2,\\"column\\":1,\\"remoteUrl\\":\\"filename1.ts#L2\\"}",
  "2": "{\\"filename\\":\\"filename2.ts\\",\\"line\\":7,\\"column\\":4,\\"remoteUrl\\":\\"filename2.ts#L7\\"}",
  "3": "{\\"filename\\":\\"filename3.ts\\",\\"line\\":1,\\"column\\":8,\\"remoteUrl\\":\\"filename3.ts#L1\\"}",
  "__basedir": "/home/user/app/packages/",
}
`;
